syntax = "proto3";

package ozonmp.ise_apartment_api.v1;

import "validate/validate.proto";
import "google/api/annotations.proto";
import "google/protobuf/timestamp.proto";

option go_package = "github.com/ozonmp/ise-apartment-api/pkg/ise-apartment-api;ise_apartment_api";

// IseApartmentApiService - Service for working with apartments
service IseApartmentApiService {
  // DescribeApartmentV1 - Describe a apartment
  rpc DescribeApartmentV1(DescribeApartmentV1Request) returns (DescribeApartmentV1Response) {
    option (google.api.http) = {
      get: "/v1/apartments/{apartment_id}"
    };
  }
}

message Apartment {
  uint64 id  = 1;
  uint64 foo = 2;
  google.protobuf.Timestamp created = 3;
}

message DescribeApartmentV1Request {
  uint64 apartment_id = 1 [(validate.rules).uint64.gt = 0];
}

message DescribeApartmentV1Response {
  Apartment value = 1;
}
